### PR TITLE
Add CLI flags to override images and kubelet binary

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

Allow overriding images and kubelet binary path from CLI or config file

#### Types of Changes ####

* New feature

#### Verification ####

* `rke2 --help`  -- look for new CLI flags
* `rke2 server --kubelet-path=X`  -- after downloading custom kubelet to X
* `rke2 server --kube-apiserver-image=X` -- where X is an alternative apiserver Docker image name

#### Linked Issues ####

#542

### CODE REVIEW

File: pkg/podexecutor/staticpod.go [hunk 1]
```diff
@@ -59,7 +60,7 @@ func (s *StaticPodConfig) Kubelet(args []string) error {
 	}
 	go func() {
 		for {
-			cmd := exec.Command("kubelet", args...)
+			cmd := exec.Command(s.KubeletPath, args...)
```
Comment (galal-hussein) [inline pkg/podexecutor/staticpod.go:63]: is that defaulted to `kubelet`?

Comment (brandond) [reply pkg/podexecutor/staticpod.go:63]: it is now!

File: pkg/images/images.go [hunk 2]
```diff
@@ -48,17 +40,14 @@ func override(defaultValue string, overrideValue string) string {
...
-	return Images{
-		SystemDefaultRegistry: systemDefaultRegistry,
-		Runtime:               override(override(dockerRegistry, systemDefaultRegistry)+"/rancher/"+RuntimeImageName+":"+strings.ReplaceAll(version.Version, "+", "-"), runtime),
-		KubeAPIServer:         override(override(dockerRegistry, systemDefaultRegistry)+"/rancher/hardened-kubernetes:"+KubernetesVersion, apiServer),
-		KubeControllManager:   override(override(dockerRegistry, systemDefaultRegistry)+"/rancher/hardened-kubernetes:"+KubernetesVersion, controllerManager),
-		KubeScheduler:         override(override(dockerRegistry, systemDefaultRegistry)+"/rancher/hardened-kubernetes:"+KubernetesVersion, scheduler),
-		ETCD:                  override(override(dockerRegistry, systemDefaultRegistry)+"/rancher/hardened-etcd:"+EtcdVersion, etcd),
-		Pause:                 override(override(dockerRegistry, systemDefaultRegistry)+"/rancher/pause:"+PauseVersion, pause),
-	}
+// Updates the image list, honoring the SystemDefaultRegistry and Image overrides if they are not empty.
```
Comment (briandowns) [inline pkg/images/images.go:43]:
```suggestion
// SetDefaults updates the image list, honoring the SystemDefaultRegistry and Image overrides if they are not empty.
```

**Review by briandowns — APPROVED** [review]
LGTM

**Review by galal-hussein — APPROVED** [review]
LGTM

**Review by dweomer — APPROVED** [review]

**Pull request merged**